shared/src/main/scala/com/comcast/ip4s/Dns.scala

```diff
@@ -45,34 +40,30 @@ trait Dns[F[_]] {
     */
   def resolveAll(hostname: Hostname): F[List[IpAddress]]
 
-  /** Gets an IP address representing the loopback interface. */
-  def loopback: F[IpAddress]
-}
+  /** Reverses the supplied address to a hostname using the platform DNS resolver.
+    *
+    * If the address cannot be reversed, the effect fails with an `UnknownHostException`.
```

Comment:
I wasn't sure about this since its a slight abuse, but I don't see why `UnknownHostException` couldn't apply here as well.

For precedence, Node.js returns the same `ENOTFOUND` error for both `resolve` and `reverse`.

Reply:
Btw, JVM instead of returning an error for `reverse` just returns the IP address as a `String` ...